collect_app/src/test/java/org/odk/collect/android/utilities/AppStateProviderTest.kt

```diff
@@ -0,0 +1,50 @@
+package org.odk.collect.android.utilities
+
+import android.content.pm.PackageInfo
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.odk.collect.android.preferences.keys.MetaKeys
+import org.odk.collect.shared.Settings
+
+class AppStateProviderTest {
+    private lateinit var appStateProvider: AppStateProvider
+    private lateinit var packageInfo: PackageInfo
+    private lateinit var metaSettings: Settings
+
+    @Before
+    fun setup() {
+        packageInfo = PackageInfo()
+        metaSettings = mock(Settings::class.java)
+        appStateProvider = AppStateProvider(packageInfo, metaSettings)
+    }
+
+    @Test
+    fun `When app is newly installed isFreshInstall() should return true when opened for the first time`() {
+        packageInfo.firstInstallTime = 100
+        packageInfo.lastUpdateTime = 100
+        `when`(metaSettings.contains(MetaKeys.FIRST_LAUNCH)).thenReturn(false)
```

Comment:
We can start using `mockito-kotlin` now to avoid the back ticks: https://github.com/mockito/mockito-kotlin/wiki/Mocking-and-verifying

Reply:
Done.